internal/orgcollector/orgcollector.go

```diff
@@ -0,0 +1,149 @@
+// NOTE: whenever the CC API V3 exposes the quota information for us,
+// we should just store everything in the app cache and get it from there
+package orgcollector
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-firehose-nozzle/internal/client/cloudfoundry"
+	"github.com/DataDog/datadog-firehose-nozzle/internal/config"
+	"github.com/DataDog/datadog-firehose-nozzle/internal/metric"
+	"github.com/DataDog/datadog-firehose-nozzle/internal/util"
+
+	"github.com/cloudfoundry-community/go-cfclient"
+	"github.com/cloudfoundry/gosteno"
+)
+
+type OrgCollector struct {
+	cfClient         *cloudfoundry.CFClient
+	log              *gosteno.Logger
+	processedMetrics chan<- []metric.MetricPackage
+	customTags       []string
+	queryInterval    uint32
+	stopper          chan bool
+}
+
+func NewOrgCollector(
+	config *config.Config,
+	processedMetrics chan<- []metric.MetricPackage,
+	log *gosteno.Logger,
+	customTags []string) (*OrgCollector, error) {
+	cfClient, err := cloudfoundry.NewClient(config, log)
+	if err != nil {
+		return nil, err
+	}
+	return &OrgCollector{
+		cfClient:         cfClient,
+		log:              log,
+		processedMetrics: processedMetrics,
+		customTags:       customTags,
+		queryInterval:    config.OrgDataQuerySeconds,
```

Comment:
This config name is confusing to me. Shouldn't it be better to call it `OrgDataCollectionInterval` or something that makes it obvious it is an interval?

Reply:
You're right, that does sound better. I'll change it.